image_writers_guide/sti.adoc

```diff
@@ -10,70 +10,52 @@
 
-The build process consists of three fundamental elements which are combined into
-final docker image, the three are: sources, scripts and builder image. During the
-build process sti must place sources and scripts inside that builder image. To do
-so sti creates a tar file containing the two and then streams that file into the
-builder image. Before executing `assemble` script, sti untars that file and places
-its contents into the location specified with `--location` flag or `STI_LOCATION`
-environment variable from the builder image (default location is `/tmp`). For this
-to happen your image must supply tar archiving utility (command `tar` available in `$PATH`)
-and command line interpreter (command `/bin/sh`). Doing so will allow your image to
-use the fastest possible build path, because in all other cases when either
-`tar` or `/bin/sh` is not available, sti build will be forced to perform an additional
-docker build to put both sources and scripts inside the image and only then run the
-usual sti build procedure (sti will do this automatically if tar and /bin/sh are not found).
-See the following diagram for how build workflow looks like:
+See the following diagram for the basic STI build workflow:
 
-image:sti-flow.png[STI workflow]
+.Build workflow
+image::sti-flow.png[STI workflow]
 
-\* Run build's responsibility is to untar the sources, scripts and artifacts (if such
-exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh`
-error) it's responsible only for invoking `assemble` script, since both scripts and
-sources are already there.
+////
+* Run build's responsibility is to untar the sources, scripts and artifacts (if such exist) and invoke `assemble` script. If this is second run (after catching `tar`/`/bin/sh` error) it's responsible only for invoking `assemble` script, since both scripts and sources are already there.
+////
 
+== STI Scripts
+STI expects you to supply the following scripts:
 
-== STI scripts
+* link:#assemble[`assemble`] (required)
+* link:#run[`run`] (required)
+* link:#save-artifacts[`save-artifacts`] (optional)
+* link:#usage[`usage`] (optional)
 
-STI expects you, as the builder image author, to supply the following scripts:
+All of the scripts can be written in any programming language, as long as it is executable inside the builder image.
 
-* required:
-    * link:[#assemble]
-    * link:[#run]
-* optional:
-    * link:[#save-artifacts]
-    * link:[#usage]
+NOTE: The examples in the following sections are written in Bash and it is assumed all tar contents are unpacked into the [filename]#/tmp/sti# directory.
 
-All of the scripts can be written in any programming language, as long as it is
-executable inside the builder image.
+[[assemble]]
+*`assemble`*
 
```

Comment:
[Suggestion]
Rather than using our commands or scripts as heading titles, maybe we can rename this to Assemble Script, and so on.